### PR TITLE
feat(CL): x/concentrated-liquidity module new (merge to main 5)

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #XXX

## What is the purpose of the change

We would like to drop `concentrated-liquidity-main` branch to avoid branch synching overhead. As part of that, we are going to be incrementally merging the current concentrated liquidity feature state to `main`.

All this logic has already been given a round of reviews since we treated `concentrated-liquidity-main` branch as `main` with appropriate review processes.

This particular PR does the following:
- merges `concentrated-liquidity` module

For the latest state of the feature that we are merging, see: https://github.com/osmosis-labs/osmosis/tree/concentrated-liquidity-main

See spec: https://github.com/osmosis-labs/osmosis/pull/3052/files

### CODE REVIEW

File: networks/osmosis-1/upgrades/v13/mainnet/guide.md [hunk 1]
```diff
@@ -1,127 +0,0 @@
-# v12 to v13 Mainnet Upgrade Guide
```
Comment (p0mvn) [inline networks/osmosis-1/upgrades/v13/mainnet/guide.md:1]: Let's keep this

File: osmomath/math.go [hunk 2]
```diff
@@ -171,3 +172,12 @@ func PowApprox(base sdk.Dec, exp sdk.Dec, precision sdk.Dec) sdk.Dec {
 	}
 	return sum
 }
+
+// Max returns the maximum value between two Ordered types.
+func Max[T constraints.Ordered](x, y T) T {
```
Comment (p0mvn) [inline osmomath/math.go:177]: Please make sure that this is not a duplicate of a function that was moved

File: x/ibc-rate-limit/ibc_middleware_test.go [hunk 3]
```diff
@@ -506,6 +506,6 @@ func (suite *MiddlewareTestSuite) TestUnsetRateLimitingContract() {
 	osmosisApp := suite.chainA.GetOsmosisApp()
 	paramSpace, ok := osmosisApp.AppKeepers.ParamsKeeper.GetSubspace(types.ModuleName)
 	suite.Require().True(ok)
-        // N.B.: this panics if validation fails.
+	// N.B.: this panics if validation fails.
```
Comment (p0mvn) [inline x/ibc-rate-limit/ibc_middleware_test.go:509]: Can't comment on that file but please ever the rate limiter wasm

File: x/swaprouter/msg_server.go [hunk 4]
```diff
@@ -56,6 +56,16 @@ func (server msgServer) CreateStableswapPool(goCtx context.Context, msg *stables
...
+// func (server msgServer) StableSwapAdjustScalingFactors(goCtx context.Context, msg *stableswap.MsgStableSwapAdjustScalingFactors) (*stableswap.MsgStableSwapAdjustScalingFactorsResponse, error) {
+// 	ctx := sdk.UnwrapSDKContext(goCtx)
+
+// 	if err := server.keeper.SetStableSwapScalingFactors(ctx, msg.ScalingFactors, msg.PoolID, msg.ScalingFactorGovernor); err != nil {
+// 		return nil, err
+// 	}
+
+// 	return &stableswap.MsgStableSwapAdjustScalingFactorsResponse{}, nil
+// }
+
```
Comment (p0mvn) [inline x/swaprouter/msg_server.go:68]:
```suggestion
```

File: x/swaprouter/types/routes.go [hunk 5]
```diff
@@ -9,16 +9,16 @@ import (
 	gammtypes "github.com/osmosis-labs/osmosis/v13/x/gamm/types"
 )
 
-// AccountI defines the account contract that must be fulfilled when
-// creating a x/swaprouter keeper.
+// AccountKeeper defines the account contract that must be fulfilled when
+// creating a x/gamm keeper.
```
Comment (p0mvn) [inline x/swaprouter/types/routes.go:13]:
```suggestion
// AccountI defines the account contract that must be fulfilled when
// creating a x/swaprouter keeper.
```

File: x/swaprouter/types/routes.go [hunk 6]
```diff
@@ -9,16 +9,16 @@ import (
...
 type AccountI interface {
 	NewAccount(sdk.Context, authtypes.AccountI) authtypes.AccountI
 	GetAccount(ctx sdk.Context, addr sdk.AccAddress) authtypes.AccountI
 	SetAccount(ctx sdk.Context, acc authtypes.AccountI)
 }
 
-// BankI defines the banking contract that must be fulfilled when
-// creating a x/swaprouter keeper.
+// BankKeeper defines the banking contract that must be fulfilled when
```
Comment (p0mvn) [inline x/swaprouter/types/routes.go:20]:
```suggestion
// BankI defines the banking contract that must be fulfilled when
```

File: x/swaprouter/types/routes.go [hunk 7]
```diff
@@ -9,16 +9,16 @@ import (
...
+// creating a x/gamm keeper.
```
Comment (p0mvn) [inline x/swaprouter/types/routes.go:21]:
```suggestion
// creating a x/swaprouter keeper.
```

**Review by p0mvn — APPROVED** [review]

File: x/ibc-rate-limit/ibc_middleware_test.go (hunk 3 above)
Comment (czarcas7ic) [reply x/ibc-rate-limit/ibc_middleware_test.go:509]: Sorry I don't quite understand what you mean here

File: osmomath/math.go (hunk 2 above)
Comment (czarcas7ic) [reply osmomath/math.go:177]: We are currently using this func here https://github.com/osmosis-labs/osmosis/blob/5d72afb0cd14a3ba5916d0f084def0075a528e3b/x/concentrated-liquidity/tick.go#L49-L65

File: x/ibc-rate-limit/ibc_middleware_test.go (hunk 3 above)
Comment (p0mvn) [reply x/ibc-rate-limit/ibc_middleware_test.go:509]: > Sorry I don't quite understand what you mean here

Please remove `rate_limiter.wasm`

What I meant is that I could not comment under that file so the comment location is wrong / confusing

**Pull request closed without merging**